app/uk/gov/hmrc/mtdsatestsupportapi/controllers/requestParsers/validators/CreateBusinessValidator.scala

```diff
@@ -0,0 +1,117 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.mtdsatestsupportapi.controllers.requestParsers.validators
+
+import api.controllers.requestParsers.validators.Validator
+import api.controllers.requestParsers.validators.validations._
+import api.models.errors._
+import play.api.libs.json.{JsDefined, JsLookupResult, JsValue}
+import uk.gov.hmrc.mtdsatestsupportapi.controllers.requestParsers.validators.validations._
+import uk.gov.hmrc.mtdsatestsupportapi.models.request.createBusiness.{Business, CreateBusinessRawData}
+
+import java.time.{Clock, LocalDate}
+import javax.inject.{Inject, Singleton}
+import scala.math.Ordered.orderingToOrdered
+import scala.util.{Failure, Success, Try}
+
+@Singleton
+class CreateBusinessValidator @Inject() (clock: Clock) extends Validator[CreateBusinessRawData] {
+
+  override def validate(data: CreateBusinessRawData): List[MtdError] =
+    (for {
+      _ <- parameterFormatValidation(data)
+      _ <- bodyEnumValidation(data.body)
+      _ <- bodyFormatValidation(data)
+      _ <- bodyFieldValidation(data)
+    } yield ()).swap.getOrElse(Nil)
+
+  private def parameterFormatValidation(data: CreateBusinessRawData): Either[List[MtdError], Unit] = {
+    val errors = NinoValidation.validate(data.nino)
+
+    errorsResult(errors)
+  }
+
+  private def bodyEnumValidation(body: JsValue): Either[List[MtdError], Unit] = {
+    val errors = validateAccountingType(body) ++ validateTypeOfBusiness(body) ++
+      validateLatencyIndicator(body, "/latencyDetails/latencyIndicator1") ++
+      validateLatencyIndicator(body, "/latencyDetails/latencyIndicator2")
+
+    errorsResult(errors)
+  }
+
+  private def bodyFormatValidation(data: CreateBusinessRawData): Either[List[MtdError], Unit] = {
+    val errors = JsonFormatValidation.validate[Business](data.body) match {
+      case Nil          => NoValidationErrors
+      case schemaErrors => schemaErrors
+    }
+
+    errorsResult(errors)
+  }
+
+  private def bodyFieldValidation(data: CreateBusinessRawData): Either[List[MtdError], Unit] = {
+    val business = data.body.as[Business]
+
+    val errors =
+      business.businessAddressCountryCode.map(CountryCodeValidation.validate(_)).getOrElse(Nil) ++
+        validatePostcode(business) ++
+        validateMissingPostcode(business) ++
+        business.firstAccountingPeriodStartDate.map(validateDate(_, "/firstAccountingPeriodStartDate")).getOrElse(Nil) ++
+        business.firstAccountingPeriodEndDate.map(validateDate(_, "/firstAccountingPeriodEndDate")).getOrElse(Nil) ++
+        business.latencyDetails.map(ld => validateDate(ld.latencyEndDate, "/latencyDetails/latencyEndDate")).getOrElse(Nil) ++
```

Comment:
Should we have validation for the `latencyDetails.taxYear1` and `latencyDetails.taxYear2`? I know there isn't an error to accommodate the failed validation of YYY-YY in the tech spec (can only see FORMAT_DATE for YYYY-MM-DD), but should we?

Reply:
Yes. Good spot!